src/theory/theory_engine.cpp

```diff
@@ -373,16 +373,20 @@ void TheoryEngine::check(Theory::Effort effort) {
 #ifdef CVC5_FOR_EACH_THEORY_STATEMENT
 #undef CVC5_FOR_EACH_THEORY_STATEMENT
 #endif
-#define CVC5_FOR_EACH_THEORY_STATEMENT(THEORY)                      \
-  if (theory::TheoryTraits<THEORY>::hasCheck                        \
-      && isTheoryEnabled(THEORY))                       \
-  {                                                                 \
-    theoryOf(THEORY)->check(effort);                                \
-    if (d_inConflict)                                               \
-    {                                                               \
-      Trace("conflict") << THEORY << " in conflict. " << std::endl; \
-      break;                                                        \
-    }                                                               \
+#define CVC5_FOR_EACH_THEORY_STATEMENT(THEORY)                           \
+  if (theory::TheoryTraits<THEORY>::hasCheck && isTheoryEnabled(THEORY)) \
+  {                                                                      \
+    theoryOf(THEORY)->check(effort);                                     \
+    if (d_inConflict)                                                    \
+    {                                                                    \
+      Trace("conflict") << THEORY << " in conflict. " << std::endl;      \
+      break;                                                             \
+    }                                                                    \
+    if (rm->out())                                                       \
```

Comment:
Can we just add this to the end of `Theory::check`?  This way it would not be run in trivial cases, e.g. for theories that have no facts.  

This might require adding `interrupt` to OutputChannel.

Reply:
We'd still need to check here whether we got interrupted.

Reply:
Hm, okay.